maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/DescriptorGeneratorMojo.java

```diff
@@ -48,8 +54,14 @@
 import org.apache.maven.tools.plugin.scanner.MojoScanner;
 import org.codehaus.plexus.component.repository.ComponentDependency;
 import org.codehaus.plexus.util.ReaderFactory;
+import org.codehaus.plexus.util.io.CachingOutputStream;
+import org.codehaus.plexus.util.io.CachingWriter;
+import org.objectweb.asm.*;
```

Comment:
Is this necessary?

Reply:
The imports are used to generate the factories for Maven 4.x mojos.  They are not using plexus, as they are booted by the new DI framework in Maven.  But this framework recognises beans annotated with `@org.apache.maven.api.di.Named`, which is not the case for mojos.  To avoid forcing the user to add this annotation, a factory (annotated with the needed annotation) is generated by the plugin.  And to avoid having to do another compile phase, the byte code is directly generated instead of generating source code.  Those new imports are needed for this feature.

Reply:
Accepted.